client/src/components/RegisterModal.vue

```diff
@@ -0,0 +1,230 @@
+<script setup lang="ts">
+import { reactive, defineEmits } from 'vue'
+
+const formData = reactive({
+  username: '',
+  email: '',
+  firstName: '',
+  lastName: '',
+  dateOfBirth: '',
+  genderId: '',
+  indigenousTIS: '',
+  password: '',
+  confirmPassword: '',
+})
+
+defineEmits(['close'])
+</script>
+
+<template>
+  <div class="register-container">
```

Comment:
The modal is very narrow for me on mobile. If you use Vuetify's ``v-dialog`` component, you can use the ``fullscreen`` prop to make the modal fullscreen on mobile.